### PR TITLE
fix: POC assigning in opportunities

### PR DESCRIPTION
**Pull request by AdityaPimpalkar**
fixes - #865

Feedback is appreciated!

https://github.com/twentyhq/twenty/assets/13139771/39a43f27-b1ed-4a2f-863f-796cf81eb8e4

### CODE REVIEW

File: front/src/modules/people/components/PeoplePicker.tsx [hunk 1]
```diff
@@ -29,16 +34,24 @@ export function PeoplePicker({
     relationPickerSearchFilterScopedState,
   );
 
+  const currentEditableFieldEntityId = useContext(EditableFieldEntityIdContext);
```
Comment (charlesBochet) [inline front/src/modules/people/components/PeoplePicker.tsx:37]: the PeoplePicker should not be aware of EditableFieldEntityIdContext as it also can be used without this context. 
EditableFieldEntityIdContext should be used as EditableField level and and company should be passed as an option props to PeoplePicker

File: front/src/modules/people/components/PeoplePicker.tsx [hunk 2]
```diff
@@ -51,14 +64,20 @@ export function PeoplePicker({
     onSubmit(selectedPerson ?? null);
   }
 
+  let entitiesToSelect = people.entitiesToSelect;
+  if (company)
+    entitiesToSelect = people.entitiesToSelect.filter(
```
Comment (charlesBochet) [inline front/src/modules/people/components/PeoplePicker.tsx:69]: this filtering should be integrated to useFilteredSearchEntityQuery logic. If not, we could end up with no result as people.entitiesToSelect only returns 10 people that could not be part of the desired company

File: front/src/modules/search/hooks/useFilteredSearchEntityQuery.ts [hunk 3]
```diff
@@ -26,7 +26,7 @@ type ExtractEntityTypeFromQueryResponse<T> = T extends {
   ? U
   : never;
 
-const DEFAULT_SEARCH_REQUEST_LIMIT = 10;
+const DEFAULT_SEARCH_REQUEST_LIMIT = 30;
```
Comment (charlesBochet) [inline front/src/modules/search/hooks/useFilteredSearchEntityQuery.ts:29]: yep, we should avoid that, it only moves the issue a bit further

**Review by charlesBochet — COMMENTED** [review]
Thanks for the PR! Added comments :)

**Comment by lucasbordeau** [comment]
@charlesBochet If you think that it can be merged as is I think it's ok, I'll create a ticket.

**Comment by AdityaPimpalkar** [comment]
> Interesting, I think that shows the limitations of the current API.
> 
> It would be nice to create a ticket to work on a more generic API, because this implementation won't scale.

Good point @lucasbordeau 👍

**Review by charlesBochet — APPROVED** [review]

**Comment by charlesBochet** [comment]
Merging it and taking a look at the other PR. We might have to do a follow up PR on this one!
Thanks @AdityaPimpalkar

**Pull request merged**